### PR TITLE
[WIP] starts using an API based off of a current version of the CLI

### PR DESCRIPTION
**Pull request by johnnyman727**
Dependent on https://github.com/tessel/t2-cli/pull/555 (or whatever other PRs that turns into).
Dependent on firmware as well: https://github.com/tessel/t2-firmware/pull/149.

This biggest change this PR introduces is running tests that are saved within this repo rather than a branch of a the CLI repo that is 500 commits behind. That way, the relevant code stays here and we can just update the version of the CLI it points to.
- [x] Tessel LED API
- [x] Will probably need to expose an API for attaches/detaches
- [x] A lot of manual testing
- [x] Renaming/documentation/refactoring
- [x] Remove stray console logs

### CODE REVIEW

File: client/index.js [hunk 1]
```diff
@@ -43,9 +43,9 @@ var server = http.createServer(app);
 
 var verifyFile = fs.readFileSync('./tests/resources/deadbeef.hex');
 var USB_OPTS = {bytes:84, verify: verifyFile}
-var ETH_OPTS = {host: '192.168.0.1'}
+var ETH_OPTS = {host: configs.host.pingIP}
```
Comment (johnnyman727) [inline client/index.js:48]: I disliked that this was hard coded. It made testing difficult.

File: client/index.js [hunk 2]
```diff
@@ -474,63 +479,58 @@ try {
...
+      var tesselLogData = sanitizeTessel(tessel);
+      var tesselData = {test: 'name', tessel: tesselLogData, status: 1, info: tesselLogData.name};
+      // Emit that we have a connected Tessel
+      io.sockets.emit("addTesselUSB", tesselLogData);
+      // Update our own log on the testlator server
+      reportLog({device: tesselLogData.serialNumber, data: "Connected over USB"}, true);
+      // Update out through hole tests interface
+      updateDeviceStatus(tesselData, true);
+
+      // tessel.setRedLED(0);
```
Comment (johnnyman727) [inline client/index.js]: I still need to add the LED API back in.

File: client/package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "mongojs": "~0.18.2",
     "request": "^2.57.0",
     "socket.io": "~0.9.14",
-    "t2": "git://github.com/tessel/t2-cli.git#jh-rig-test",
+    "t2": "git://github.com/tessel/t2-cli.git#jon-rig-improvements",
```
Comment (johnnyman727) [inline client/package.json:12]: This should point to a solid released version of the CLI (probably 0.0.11) by the time this is merged.

File: client/index.js (hunk 1 above)
Comment (rwaldron) [inline client/index.js:48]: :+1:

File: client/package.json (hunk 3 above)
Comment (rwaldron) [inline client/package.json:12]: :+1:

**Comment by johnnyman727** [comment]
I added a USB Tessel scanning library (/tests/discovery/usb_discovery.js) that is separate from the small one that the T2 CLI uses. I did this because the test rig needs some functionality that the general CLI does not need or offer:
- support for multiple scans at once
- events when new Tessels are added
- events when Tessels are removed
- No claiming of interfaces

I think this new library is actually better than the one the CLI has and we might integrate them at some point but for now they have different enough requirements to keep them apart.

**Comment by johnnyman727** [comment]
Closing so I can open a PR on master... ugh

**Comment by johnnyman727** [comment]
Working on https://github.com/tessel/t2-test-rig-sw/pull/13 now.

**Pull request closed without merging**